app/controllers/organizations/staff/invitations_controller.rb

```diff
@@ -17,13 +17,28 @@ def create
       authorize! User, context: {organization: Current.organization},
         with: Organizations::StaffInvitationPolicy
 
+      # User may already exist as adopter, fosterer, or staff for another
+      # organization and there needs to be a way to invite them as staff
+      # for this organization.
+      # TODO: The user should probably get an email letting them know they
+      # were added to a new organization.
+      if User.where(email: user_params[:email]).any?
+        @user = User.find_by(email: user_params[:email])
+        if !@user.has_role?(user_params[:roles], Current.organization)
+          @user.add_role(user_params[:roles], Current.organization)
+          redirect_to staff_staff_index_path, notice: t(".success")
+          return
+        end
+      end
```

Comment:
Thoughts on this flow instead?

Keep the comments as is 
```suggestion
     @user = User.find_by(email: user_params[:email]
     
     if @user.present?
        if !@user.has_role?(user_params[:roles], Current.organization)
          @user.add_role(user_params[:roles], Current.organization)
          redirect_to staff_staff_index_path, notice: t(".success")
          return
        end
     else
     .
     .
     .
   end
```

Reply:
I don't think the else is needed. I prefer returning vs wrapping the rest of the function in a large else block. I get the following error if I don't include the return statement so will keep it for now (unless you are aware of another work around).

> AbstractController::DoubleRenderError: Render and/or redirect were called multiple times in this action. Please note that you may only call render OR redirect, and at most once per action. Also note that neither redirect nor render terminate execution of the action, so if you want to exit an action after redirecting, you need to do something like "redirect_to(...); return".